Sources/PrimerSDK/Classes/Core/Checkout Components/PrimerHeadlessUniversalCheckoutUIManager.swift

```diff
@@ -184,68 +191,122 @@ extension PrimerHeadlessUniversalCheckout {
         }
         
         private func buildRequestBody() -> Promise<Request.Body.Tokenization> {
+            switch self.paymentMethodType {
+            case PrimerPaymentMethodType.paymentCard.rawValue:
+                return makeCardRequestBody()
+            case PrimerPaymentMethodType.adyenBancontact.rawValue:
+                return makeCardRedirectRequestBody()
+            default:
+                fatalError()
```

Comment:
Please do this a `precondition` and return an error to the delegate

Reply:
I've made the function to `throw` an error in case of an unsupported payment method type so that the chain is interrupted